src/commands/coverage/__tests__/upload.test.ts

```diff
@@ -0,0 +1,345 @@
+        'src/commands/coverage/__tests__/fixtures/another-jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/subfolder/nested-jacoco-report.xml',
+      ])
+    }, 10000)
+    test('should fetch nested folders and ignore non xml files', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const result = command['getMatchingCoverageReportFilesByFormat'].call({
+        basePaths: ['**/coverage/**'],
+        config: {},
+        context,
+      })
+
+      const fileNames = Object.values(result).flatMap((paths) => paths)
+
+      expect(fileNames).toEqual([
+        'src/commands/coverage/__tests__/fixtures/another-jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/jacoco-report.xml',
+        'src/commands/coverage/__tests__/fixtures/subfolder/nested-jacoco-report.xml',
+      ])
+    }, 10000)
+  })
+  describe('getSpanTags', () => {
+    test('should parse DD_ENV environment variable', async () => {
+      process.env.DD_ENV = 'ci'
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const spanTags: SpanTags = await command['getSpanTags'].call({
+        config: {
+          env: process.env.DD_ENV,
+        },
+        context,
+      })
+      expect(spanTags).toMatchObject({
+        env: 'ci',
+      })
+    })
+  })
+  describe('parseCustomTags', () => {
+    test('should parse tags argument', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const spanTags: SpanTags = command['getCustomTags'].call({
+        config: {},
+        context,
+        tags: ['key1:value1', 'key2:value2'],
+      })
+
+      expect(spanTags).toMatchObject({
+        key1: 'value1',
+        key2: 'value2',
+      })
+    })
+    test('should parse DD_TAGS environment variable', async () => {
+      process.env.DD_TAGS = 'key1:https://google.com,key2:value2'
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const spanTags: SpanTags = command['getCustomTags'].call({
+        config: {
+          envVarTags: process.env.DD_TAGS,
+        },
+        context,
+      })
+      expect(spanTags).toMatchObject({
+        key1: 'https://google.com',
+        key2: 'value2',
+      })
+    })
+    test('should parse measures argument', async () => {
+      const context = createMockContext()
+      const command = new UploadCodeCoverageReportCommand()
+      const spanTags: SpanTags = command['getCustomMeasures'].call({
+        config: {},
+        context,
+        measures: ['key1:10', 'key2:20'],
+      })
+
+      expect(spanTags).toMatchObject({
+        key1: 10,
+        key2: 20,
+      })
+    })
+    test('should parse DD_MEASURES environment variable', async () => {
```

Comment:
do we want to check what happens if you pass a non numeric value?

Reply:
Makes sense, added a test case for this